fenix/app/src/main/java/org/mozilla/fenix/shopping/middleware/ReviewQualityCheckNetworkMiddleware.kt

```diff
@@ -116,4 +145,26 @@ class ReviewQualityCheckNetworkMiddleware(
     ) {
         dispatch(ReviewQualityCheckAction.UpdateProductReview(productReviewState))
     }
+
+    private fun Store<ReviewQualityCheckState, ReviewQualityCheckAction>.restoreAnalysingStateIfRequired(
+        productReviewState: ProductReviewState,
+        productAnalysis: GeckoProductAnalysis?,
+        productPageUrl: String?,
+    ) {
+        if (productPageUrl == null) {
+            return
+        }
+
+        val productsAnalysed = appStore.state.shoppingState.productsAnalysed
+        if (productReviewState.isAnalysisPresentOrNoAnalysisPresent() &&
+            productAnalysis?.needsAnalysis == true &&
+            productsAnalysed.contains(productPageUrl)
+        ) {
```

Comment:
nit: you may be able to consolidate the ifs

```suggestion
        if (productPageUrl == null) {
            return
        }

        val productsAnalysed = appStore.state.shoppingState.productsAnalysed
        if (productPageUrl != null && 
            productReviewState.isAnalysisPresentOrNoAnalysisPresent() &&
            productAnalysis?.needsAnalysis == true &&
            productsAnalysed.contains(productPageUrl)
        ) {
```
OR keep the original and `if` and now do:
```suggestion
        productPageUrl?.let {
        ...
        }
```